### DISCUSSION TITLE
Fail to create "latest" symlink and not delete old backups

### DISCUSSION THREAD

**Discussion by colombarir** [prodrigestivill/docker-postgres-backup-local#124] [Q&A]
Hello, in my config I specified these params
```
BACKUP_KEEP_MINS=1440
BACKUP_KEEP_DAYS=7
BACKUP_KEEP_WEEKS=4
BACKUP_KEEP_MONTHS=6
```
nevertheless, in the container's logs I find these messages

```
2023/02/06 21:30:49 148098: Point last backup file to this last backup...
2023/02/06 21:30:49 148098: ln: failed to create symbolic link '/backups/last/xxx-latest.sql.gz'
2023/02/06 21:30:49 148098: : Permission denied
2023/02/06 21:30:49 148098 Exit Status: 1
(0x2e0128,0x423370)
2023/02/06 23:28:50 Stopping
2023/02/06 23:28:50 Waiting
2023/02/06 23:28:50 Exiting
2023/02/06 23:28:52 Running version: v0.0.10
```
Moreover, Last, daily, weekly and monthly files are regularly created, but the older aren't deleted.

Can someone help me to fix this problem?

**Comment by prodrigestivill** [comment]
Did you follow the usage instructions from the README.md?

Specially the procedure regarding:
_In case of running as postgres user, the system administrator must initialize the permission of the destination folder as follows..._

**Reply by colombarir** [reply]
Thank you for the suggestion, I tried to do it but the destination folder is a shared folder on a NAS on which I don't have permission to change owners and groups. 

Anyway, the folder (and its subfolders) has read-write-execute permission to all... al least the deletion of old files should be possible, right?

**Reply by prodrigestivill** [reply]
Check the filesystem on the NAS support hardlinks and softlinks. Also if you have nfs limited to a single user, please configure the docker or docker-compose to use that specific uid and gid.

**Comment by madkatz01** [comment]
I am also struggling with this and similarly my backup folder is a folder on a shared NAS, mounted with CIFS. But the local Ubuntu server has all the read/write privileges for that shared folder, so not sure why it's failing. Other containers can read/write just fine to that shared drive. Filesystem on NAS is btrfs so I know it supports hardlinks. My error is slightly different though (says operation not supported instead of permission denied).

```
Point last backup file to this last backup...
ln: failed to create symbolic link '/db_dumps/last/xxx-latest.sql.gz': Operation not supported
```

**Reply by prodrigestivill** [reply]
Same answer as #97, the docker volume needs to support hard links and soft links.

Also added more information in the README.md (187e2e2f6e34566c8a850f06c0f93b319fcc3e5b).

If you need to use a NAS for that volume, please use NFS or 9P.

**Reply by madkatz01** [reply]
Thanks for the clarification. I see that it was the CIFS protocol interfering, even though the host NAS has a compatible filesystem. Got it all working now!
